### PR TITLE
Use scala 2.13

### PR DESCRIPTION
**Pull request by melgenek**
Update library versions to the latest ones.
Remove support for 2.11, because circe does not support 2.11 in latest version.
Remove unused cats imports.

### CODE REVIEW

**Comment by melgenek** [comment]
Hi! I can definitely split this pr into multiple prs.
I would also keep 2.11 scala. The problem is circle libary, that drops 2.11 support in `0.12.0-M4`. I read through circe releases and it does not seems that there were any truly great upgrates since `0.12.0-M3` except some optimizations and bug fixes. So if you don't mind having [`0.12.0-M3`](https://github.com/circe/circe/releases/tag/v0.12.0-M3) that supports `2.11/2.12/2.13` as a dependency, then I am glad to use it.

**Comment by chuwy** [comment]
Oh, I see. I completely missed the fact that circe dropped 2.11. In that case I guess we can drop it from here circe core as well. Just in case - could you please delete it in the "Bump circe to 0.12.1" commit - I'll add a note there.

File: project/Dependencies.scala [hunk 1]
```diff
@@ -15,10 +15,10 @@ import sbt._
 object Dependencies {
   object V {
     // Scala
-    val json4s          = "3.2.11"
-    val circe           = "0.11.1"
-    val cats            = "1.6.0"
-    val specs2          = "4.5.1"
+    val json4s          = "3.6.7"
+    val circe           = "0.12.1"
```
[the remark this answers is not in the document]
Comment (melgenek) [reply project/Dependencies.scala]: PR: https://github.com/snowplow-incubator/iglu-scala-core/pull/38

File: project/Dependencies.scala [hunk 2]
```diff
@@ -15,10 +15,10 @@ import sbt._
 object Dependencies {
   object V {
     // Scala
-    val json4s          = "3.2.11"
-    val circe           = "0.11.1"
-    val cats            = "1.6.0"
-    val specs2          = "4.5.1"
+    val json4s          = "3.6.7"
```
[the remark this answers is not in the document]
Comment (melgenek) [reply project/Dependencies.scala]: PR: https://github.com/snowplow-incubator/iglu-scala-core/pull/41

**Comment by melgenek** [comment]
There is a pr for specs2 version bump https://github.com/snowplow-incubator/iglu-scala-core/pull/42, which fails unexpectedly with scala 2.11. While specs2:2.5.1 supports scala 2.13, this upgrade is not really needed.

I would merge https://github.com/snowplow-incubator/iglu-scala-core/pull/38, https://github.com/snowplow-incubator/iglu-scala-core/pull/39, https://github.com/snowplow-incubator/iglu-scala-core/pull/41 first so that it is possible to base this pr on the master.

**Comment by chuwy** [comment]
Cherry-picked into https://github.com/snowplow-incubator/iglu-scala-core/pull/29. Thanks @melgenek!

**Pull request closed without merging**